### PR TITLE
Add new properties to the metadata API

### LINKED ISSUE #3557: Add missing properties to the metadata API - Subscriptions, runtimeVersion, appConnectionProperties
**What content needs to be created or modified?**

There are some new properties being introduced in the metadata API, add them to the API reference docs.
1. runtimeVersion
2. appConnectionProperties
3. Subscriptions

**Where should the new material be placed?**

https://docs.dapr.io/reference/api/metadata_api/

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/dapr/pull/6524
https://github.com/dapr/dapr/pull/6537

Missing subscription information from this 1.10 issue https://github.com/dapr/dapr/issues/5510

**Additional context**

### PR DESCRIPTION
**Pull request by shubham1172**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This PR adds the following to metadata API options

1. runtimeVersion
1. appConnectionProperties
1. subscriptions

## Issue reference

This PR will close #3557

### CODE REVIEW

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 1]
```diff
@@ -17,12 +19,21 @@ Component type | Capabilities
...
-## HTTPEndpoints
+### HTTPEndpoints
 Each loaded `HttpEndpoint` provides a name to easily identify the Dapr resource associated with the runtime.
 
-## Attributes
+### Subscriptions
+The metadata API returns a list of pubsub subscriptions that the app has registered with the Dapr runtime. This includes the pubsub name, topic, routes, deadletter topic, and the metadata associated with the subscription.
+
+### Enabled features
+A list of features enabled during build time, this does not include the features from Configuration spec.
```
Comment (shubham1172) [inline daprdocs/content/en/reference/api/metadata_api.md]: FYI @ItalyPaleAle

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/api/metadata_api.md]: It’s actually the features from the config spec!

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/api/metadata_api.md]: I just realized I may have picked the wrong variable. The intent was to have those from configurarion (which does include the build time flags)

Comment (shubham1172) [reply daprdocs/content/en/reference/api/metadata_api.md]: No problem - I have converted https://github.com/dapr/dapr/pull/6537 to draft for now, I will update the docs once your changes are in.

Comment (ItalyPaleAle) [reply daprdocs/content/en/reference/api/metadata_api.md]: Done

Comment (shubham1172) [reply daprdocs/content/en/reference/api/metadata_api.md]: The original PR and docs have been updated.

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 2]
```diff
@@ -17,12 +19,21 @@ Component type | Capabilities
 State Store    | ETAG, TRANSACTION, ACTOR, QUERY_API
 Binding        | INPUT_BINDING, OUTPUT_BINDING
 
-## HTTPEndpoints
+### HTTPEndpoints
 Each loaded `HttpEndpoint` provides a name to easily identify the Dapr resource associated with the runtime.
 
-## Attributes
+### Subscriptions
+The metadata API returns a list of pubsub subscriptions that the app has registered with the Dapr runtime. This includes the pubsub name, topic, routes, deadletter topic, and the metadata associated with the subscription.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/metadata_api.md]:
```suggestion
The metadata API returns a list of pub/sub subscriptions that the app has registered with the Dapr runtime. This includes the pub/sub name, topic, routes, dead letter topic, and the metadata associated with the subscription.
```

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 3]
```diff
@@ -17,12 +19,21 @@ Component type | Capabilities
...
+### Enabled features
+A list of features enabled via Configuration spec (including build-time overrides).
+
+### App connection details
+The metadata API returns information related to Dapr's connection to the app. This includes the app port, protocol, host, max concurrency, along with health check details.
 
-The metadata API allows you to store additional attribute information in the format of key-value pairs. These are ephemeral in-memory and are not persisted if a sidecar is reloaded. This information should be added at the time of a sidecar creation, for example, after the application has started. 
+### Attributes
+
+The metadata API allows you to store additional attribute information in the format of key-value pairs. These are ephemeral in-memory and are not persisted if a sidecar is reloaded. This information should be added at the time of a sidecar creation, for example, after the application has started.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/metadata_api.md]:
```suggestion
The metadata API allows you to store additional attribute information in the format of key-value pairs. These are ephemeral in-memory and are not persisted if a sidecar is reloaded. This information should be added at the time of a sidecar creation (for example, after the application has started).
```

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 4]
```diff
@@ -57,9 +68,13 @@ Code | Description
...
 ----                   | ----                                                                  | -----------
 id                     | string                                                                | Application ID
+runtimeVersion         | string                                                                | Version of the Dapr runtime
 actors                 | [Metadata API Response Registered Actor](#metadataapiresponseactor)[] | A json encoded array of registered actors metadata.
 extended.attributeName | string                                                                | List of custom attributes as key-value pairs, where key is the attribute name.
 components             | [Metadata API Response Component](#metadataapiresponsecomponent)[]    | A json encoded array of loaded components metadata.
+httpEndpoints          | [Metadata API Response HttpEndpoint](#metadataapiresponsehttpendpoint)[] | A json encoded array of loaded HttpEndpoints metadata.
+subscriptions          | [Metadata API Response Subscription](#metadataapiresponsesubscription)[] | A json encoded array of pubsub subscriptions metadata.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/metadata_api.md]:
```suggestion
subscriptions          | [Metadata API Response Subscription](#metadataapiresponsesubscription)[] | A json encoded array of pub/sub subscriptions metadata.
```

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 5]
```diff
@@ -75,7 +90,50 @@ Name    | Type   | Description
...
+
+Name    | Type   | Description
+----    | ----   | -----------
+name    | string | Name of the HttpEndpoint.
+
+<a id="metadataapiresponsesubscription"></a>**Metadata API Response Subscription**
+
+Name            | Type   | Description
+----            | ----   | -----------
+pubsubname      | string | Name of the pubsub.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/metadata_api.md]:
```suggestion
pubsubname      | string | Name of the pub/sub.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
just some small edits for consistency!

**Comment by shubham1172** [comment]
@hhunter-ms thank you so much for the suggestions, merged 'em all.

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/api/metadata_api.md [hunk 6]
```diff
@@ -6,9 +6,11 @@ description: "Detailed documentation on the Metadata API"
 weight: 1100
 ---
 
-Dapr has a metadata API that returns information about the sidecar allowing runtime discoverability. The metadata endpoint returns a list of the resources (components and HttpEndpoints loaded), the activated actors (if present), and attributes with information attached.
+Dapr has a metadata API that returns information about the sidecar allowing runtime discoverability. The metadata endpoint returns the runtime version, a list of the resources (components and HttpEndpoints loaded), the activated actors (if present), subscriptions, enabled features, app connection details, and attributes with information attached.
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/metadata_api.md]: This raises the issue, of the need to return a list of the resiliency policies as well, which is missing.

```suggestion
Dapr has a metadata API that returns information about the sidecar allowing runtime discoverability. The metadata endpoint returns the following information.
- Runtime version
- List of the loaded resources (`components`, `subscriptions` and `HttpEndpoints`)
- Registered actor types
- Features enabled
- Application connection details
- Custom, ephemeral attributes with information.
```

Comment (shubham1172) [reply daprdocs/content/en/reference/api/metadata_api.md]: @msfussell interesting idea, do you have any specific usecase in mind and how much details should the metadata API expose? Just the name of the registered policies? targets?

**Comment by msfussell** [comment]
@shubham1172 - I raised this issue https://github.com/dapr/docs/issues/3566

**Comment by msfussell** [comment]
@shubham1172 - I raised this issue https://github.com/dapr/dapr/issues/6550

**Comment by shubham1172** [comment]
Thanks @msfussell, could you unassign me from https://github.com/dapr/docs/issues/3566, probably the one who picks up the dapr runtime issue will pick the docs one.

Could you also approve/merge this PR if it looks good?

**Comment by hhunter-ms** [comment]
> Thanks @msfussell, could you unassign me from #3566, probably the one who picks up the dapr runtime issue will pick the docs one.

just unassigned you - thanks!

**Pull request merged**